src/EditorFeatures/CSharpTest/Diagnostics/MakeMethodSynchronous/MakeMethodSynchronousTests.cs

```diff
@@ -0,0 +1,356 @@
+{
+    public class MakeMethodSynchronousTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override Tuple<DiagnosticAnalyzer, CodeFixProvider> CreateDiagnosticProviderAndFixer(Workspace workspace)
+        {
+            return Tuple.Create<DiagnosticAnalyzer, CodeFixProvider>(null, new CSharpMakeMethodSynchronousCodeFixProvider());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)]
+        public async Task TestTaskReturnType()
+        {
+            await TestAsync(
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    async Task [|Foo|]()
+    {
+    }
+}",
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    void Foo()
+    {
+    }
+}",
+compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)]
+        public async Task TestTaskOfTReturnType()
+        {
+            await TestAsync(
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    async Task<int> [|Foo|]()
+    {
+    }
+}",
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    int Foo()
+    {
+    }
+}",
+compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)]
+        public async Task TestSecondModifier()
+        {
+            await TestAsync(
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    public async Task [|Foo|]()
+    {
+    }
+}",
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    public void Foo()
+    {
+    }
+}",
+compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)]
+        public async Task TestFirstModifier()
+        {
+            await TestAsync(
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    async public Task [|Foo|]()
+    {
+    }
+}",
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    public void Foo()
+    {
+    }
+}",
+compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)]
+        public async Task TestTrailingTrivia()
+        {
+            await TestAsync(
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    async // comment
+    Task [|Foo|]()
+    {
+    }
+}",
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    void Foo()
+    {
+    }
+}",
+compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)]
+        public async Task TestRenameMethod()
+        {
+            await TestAsync(
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    async Task [|FooAsync|]()
+    {
+    }
+}",
+@"
+using System.Threading.Tasks;
+
+class C
+{
+    void Foo()
+    {
+    }
+}",
+compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeMethodSynchronous)]
+        public async Task TestRenameMethod1()
```

Comment:
Perhaps some additional scenarios where the non-Async method already existed as an overload? What does rename do here?